### PR TITLE
Refactor `resource` package

### PR DESCRIPTION
**Pull request by NoComment1105**
A small refactor of the `resource` package after it was suggested by @LambdAurora. Also mapped some remaining fields and methods as I was there.

### CODE REVIEW

File: mappings/net/minecraft/resource/ReloadableResourceManager.mapping [hunk 1]
```diff
@@ -7,6 +7,13 @@ CLASS net/minecraft/unmapped/C_wdclgdrj net/minecraft/resource/ReloadableResourc
 	COMMENT their {@linkplain ResourceReloader#reload reload} called.
 	COMMENT
 	COMMENT @see ResourceReloader
+	FIELD f_akqxvhyz LOGGER Lorg/slf4j/Logger;
+	FIELD f_dcgxesdf type Lnet/minecraft/unmapped/C_abwutbbk;
+	FIELD f_iddtxxpw resources Lnet/minecraft/unmapped/C_carvbxzj;
+	FIELD f_rgpmwopx listeners Ljava/util/List;
```
Comment (LambdAurora) [inline mappings/net/minecraft/resource/ReloadableResourceManager.mapping]:
```suggestion
	FIELD f_rgpmwopx reloaders Ljava/util/List;
```

**Review by LambdAurora — APPROVED** [review]

File: mappings/net/minecraft/unmapped/C_carvbxzj.mapping [hunk 2]
```diff
@@ -0,0 +1,2 @@
+CLASS net/minecraft/unmapped/C_carvbxzj
```
Comment (OroArmor) [inline mappings/net/minecraft/unmapped/C_carvbxzj.mapping]: I would delete this file

**Review by OroArmor — APPROVED** [review]

File: mappings/net/minecraft/unmapped/C_carvbxzj.mapping (hunk 2 above)
Comment (NoComment1105) [reply mappings/net/minecraft/unmapped/C_carvbxzj.mapping]: I have locally, must've forgot to push

Comment (NoComment1105) [reply mappings/net/minecraft/unmapped/C_carvbxzj.mapping]: done

File: mappings/net/minecraft/resource/ResourceFactory.mapping [hunk 3]
```diff
@@ -1,2 +1,4 @@
 CLASS net/minecraft/unmapped/C_xadhkzfq net/minecraft/resource/ResourceFactory
 	COMMENT Provides resource access.
+	METHOD getResource getResource (Lnet/minecraft/unmapped/C_ncpywfca;)Lnet/minecraft/unmapped/C_tuynegco;
+		ARG 1 identifier
```
Comment (IotaBread) [inline mappings/net/minecraft/resource/ResourceFactory.mapping]:
```suggestion
		ARG 1 id
```

**Review by IotaBread — APPROVED** [review]

**Pull request merged**